pkg/reconciler/ingress/controller.go

```diff
@@ -238,3 +252,23 @@ func getReadyIngresses(ctx context.Context, knativeClient networkingClientSet.Ne
 	}
 	return ingressesToWarm, nil
 }
+
+func readyAddresses(eps *corev1.Endpoints) sets.String {
+	var count int
+	for _, subset := range eps.Subsets {
+		count += len(subset.Addresses)
+	}
+
+	if count == 0 {
+		return nil
+	}
+
+	ready := make(sets.String, count)
```

Comment:
I wonder if it's worth it to calculate `count` in this case.

Reply:
It should be super quick because endpoints usually have at most a handful of subsets (which is what we need to iterate over to compute it).

Reply:
Yeah, in this case, I would personally just delete those lines unless I measured that this has a significant perf impact.
But it's just personal preference :+1:

Reply:
I agree with that sentiment in principle. I followed what we do in other similar places here. I'll write benchmarks for some helpers upstream though, to proof/disproof this and then adjust.